futures/build.rs

```diff
@@ -0,0 +1,3 @@
+fn main() {
+    println!("cargo:rustc-check-cfg=cfg(futures_sanitizer)");
```

Comment:
This is a cfg that is set globally by CI, so I would like to allow it across the workspace using the `[workspace.lints]` table (like [this](https://github.com/crossbeam-rs/crossbeam/blob/abf24e1f31a76ef2590688d0c2bb55f82c7410a9/Cargo.toml#L66-L69)) instead of only allowing it in a specific crate using a build script.